src/MessageCreationUtil.js

```diff
@@ -100,16 +98,22 @@ export default class MessageCreationUtil {
         return streamMessage
     }
 
-    static computeStreamPartition(partitionCount, partitionKey) {
+    hash(stringToHash) {
+        if (this.cachedHashes[stringToHash] === undefined) {
+            this.cachedHashes[stringToHash] = crypto.createHash('md5').update(stringToHash).digest()
```

Comment:
Now, `cachedHashes` can grow very large over time. Maybe instead of a plain object, store them into [some simple cache that support expiration](https://www.npmjs.com/package/node-cache)?

Reply:
Ah never mind: this is in the client, so it won't see that many different values. Somehow I was thinking this was server-side.